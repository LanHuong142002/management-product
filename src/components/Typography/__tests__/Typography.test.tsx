import { cleanup, render, screen } from '@testing-library/react';
import { Typography, TypographyProps } from '@components';

describe('Testing Typography component', () => {
  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    text: 'Management',
    weight: 'regular',
  } as TypographyProps;

  it('Should render Typography component', () => {
    const { container } = render(<Typography {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render default Typography correctly', () => {
    render(<Typography {...defaultProps} />);

    const typography = screen.getByText('Management');

    expect(typography).toBeInTheDocument();
  });

  it('Should render Typography without weight', () => {
    const { container } = render(<Typography text='Management' />);

    expect(container).toMatchSnapshot();
  });
});
