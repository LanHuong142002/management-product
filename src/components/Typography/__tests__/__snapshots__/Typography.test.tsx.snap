// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Typography component Should render Typography component 1`] = `
<div>
  <p
    class="typography typography-color-primary typography-size-md typography-weight-regular"
  >
    Management
  </p>
</div>
`;

exports[`Testing Typography component Should render Typography without weight 1`] = `
<div>
  <p
    class="typography typography-color-primary typography-size-md "
  >
    Management
  </p>
</div>
`;
